docs/ref/models/constraints.txt

```
@@ -282,27 +282,43 @@ PostgreSQL 15+.
 
 .. attribute:: UniqueConstraint.violation_error_code
 
-The error code used when ``ValidationError`` is raised during
-:ref:`model validation <validating-objects>`. Defaults to ``None``.
+The error code used when a ``ValidationError`` is raised during
+:ref:`model validation <validating-objects>`.
+
+If :attr:`~UniqueConstraint.fields` is set and without a
+:attr:`~UniqueConstraint.condition`, the
+:attr:`~django.db.models.UniqueConstraint.violation_error_code` defaults to the
+same error code as when setting :attr:`.Field.unique` when there is a single
+field, and to the same error code as when setting
+:attr:`Meta.unique_together <django.db.models.Options.unique_together>` when
+there are multiple.
 
-This code is *not used* for :class:`UniqueConstraint`\s with
-:attr:`~UniqueConstraint.fields` and without a
-:attr:`~UniqueConstraint.condition`. Such :class:`~UniqueConstraint`\s have the
-same error code as constraints defined with :attr:`.Field.unique` or in
-:attr:`Meta.unique_together <django.db.models.Options.constraints>`.
+Otherwise, defaults to :attr:`.BaseConstraint.violation_error_message`.
+
+.. versionchanged:: 5.2
+
+If :attr:`~django.db.models.UniqueConstraint.violation_error_code` is
+set, this code is now always used.
```

```suggestion
    In older versions, the provided
    :attr:`.UniqueConstraint.violation_error_code` was not used when
    :attr:`.UniqueConstraint.fields` was set.
```
We need an indent and write this in past tense